### PR TITLE
Closes #1336 add NEWS.md update for basket_select()@devel

### LINKED ISSUE #1336: Feature Request: Replace `smq_select()` and `sdg_select()` with `basket_select()`
### Feature Idea

```
basket_select(
  name = NULL,
  id = NULL,
  scope = NULL,
  type =)
```
should be implemented replacing `smq_select()` and `sdg_select()`. The `type` argument expects a character scalar. It is passed to the company specific `get_terms()` function such that the function can determine which sort of basket is requested, SMQ, SDG, or something else like AEGT at Roche. The `scope` argument should accept `"BROAD"`, `"NARROW"`, and `NA_character_`.

`create_query_data()` should be updated accordingly, i.e.,
- `meddra_version` and `whodd_version` argument should be replaced by `version` and `get_smq_fun` and `get_sdg_fun` argument by `get_terms_fun`,
- the expected arguments for the function specified for `get_terms_fun` should be changed to `type`, `basket_select`, `version`, `keep_id`, and `temp_env`.

### Relevant Input

_No response_

### Relevant Output

_No response_

### Reproducible Example/Pseudo Code

_No response_

### PR DESCRIPTION
**Pull request by tamarasenior**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/main/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/main/articles/unit_test_guidance.html#writing-unit-tests-in-admiral-)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/main/articles/programming_strategy.html#deprecation-1)?
- [x] Update to all relevant roxygen headers and examples 
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -44,6 +44,8 @@ based on [Common Terminology Criteria for Adverse Events (CTCAE) v5.0](https://c
 
 - New function `convert_na_to_blanks()` converts `NA` strings to blank while preserving all attributes (#1589) 
 
+- Replace `smq_select()` and `sdg_select()` with `basket_select()` (#1597)
```
Comment (bundfussr) [inline NEWS.md:47]: Could you move it to the "Breaking Changes" section?

And could you add the changes in `create_query_data()`, too?

**Review by bundfussr — CHANGES_REQUESTED** [review]

**Comment by bundfussr** [comment]
Depending on the feedback from Thomas and Ben we may need to add further changes. In principle we would need to deprecate the functions and arguments we have removed. But in this case it is quite complicated (see my comment in #1336).

**Comment by thomas-neitmann** [comment]
Let's deprecate the old paramter by using `deprecate_stop()` and telling the user which parameters to use instead. No need to assign the old parameters to the new ones.

**Review by bundfussr — APPROVED** [review]
Well done!

**Pull request merged**